src/parse/asp/grammar.go

```diff
@@ -42,7 +42,9 @@ type FuncDef struct {
 	// Not part of the grammar. Used to indicate internal targets that can only
 	// be called using keyword arguments.
 	KeywordsOnly bool
-	IsPrivate    bool
+	// allowed return types of the FuncDef
+	Returns   []string
```

Comment:
I wonder if this should just be a single `string`? We don't have any cases where we return different types and it would make the parser simpler to just read a single identifier.